pkg/api/util/doget.go

```diff
@@ -39,22 +38,34 @@ func GetClient(verify bool) *http.Client {
 	return GetClientWithTimeout(0, verify)
 }
 
+// ipcRoundTripper is an implementation of http.RoundTripper interface
+type ipcRoundTripper struct {
+	tr http.Transport
+}
+
+// RoundTrip implement the http.RoundTripper interface method to be used in http request context
+// It is used to lazy load the TLS Client config, in order to let the time to initialize it before using it
+func (i *ipcRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if i.tr.TLSClientConfig == nil {
+		i.tr.TLSClientConfig = GetTLSClientConfig()
+	}
+
+	return i.tr.RoundTrip(req)
+}
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏This is not thread safe, do you know if this could be called concurrently ?

Reply:
Good point !
Since the `http.Transport` is made be called concurrently, and ipcRoundTripper is only a wrapper, we should make sure it's also thread-safe.
I've updated it accordingly.